### PR TITLE
Add just-debounce-it package

### PR DESCRIPTION
**Pull request by ETNOL**

### CODE REVIEW

**Comment by ETNOL** [comment]
Fixed linted errors.

File: README.md [hunk 1]
```diff
@@ -588,6 +589,27 @@ const cubedRoot = partial(Math.pow, undefined, 1/3);
...
 ```
+### [just-debounce-it](https://www.npmjs.com/package/just-debounce-it)
+:icecream:[`Try It`](http://anguscroll.com/just/just-debounce-it)
+
+`npm install just-debounce-it`
+
+```js
+import debounce from 'just-debounce-it';
+
+var debouncedFunc = debounce(function () {
```
Comment (angus-c) [inline README.md]: const

Comment (angus-c) [inline README.md]: const

File: README.md [hunk 2]
```diff
@@ -588,6 +589,27 @@ const cubedRoot = partial(Math.pow, undefined, 1/3);
...
+    console.log('Hello World');
+}, 500);
+
+for (var i = 0; i < 5; i++) { debouncedFunc(); } // 500ms later - Hello World
```
Comment (angus-c) [inline README.md]: const

File: packages/function-debounce/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Part of a [library](../../../../) of zero-dependency npm modules that do just do one thing.  
+Guilt-free utilities for every occasion.
+
+[Try it now](http://anguscroll.com/just/just-debounce-it)
+
+```js
+import debounce from 'just-debounce-it';
+
+var debouncedFunc = debounce(function () {
```
Comment (angus-c) [inline packages/function-debounce/README.md]: const

File: packages/function-debounce/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    console.log('Hello World');
+}, 500);
+
+for (var i = 0; i < 5; i++) { debouncedFunc(); } // 500ms later - Hello World
```
Comment (angus-c) [inline packages/function-debounce/README.md]: const

File: packages/function-debounce/index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+module.exports = debounce;
+
+function debounce(func, wait, immediate) {
+  var timeout;
+  return function () {
+    var context = this, args = arguments;
+    var later = function () {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
```
Comment (angus-c) [inline packages/function-debounce/index.js]: if {
...
}

File: packages/function-debounce/index.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    };
+    var callNow = immediate && !timeout;
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+    if (callNow) func.apply(context, args);
```
Comment (angus-c) [inline packages/function-debounce/index.js]: if {

File: test/function-debounce/index.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+var test = require('tape');
+var debounce = require('../../packages/function-debounce');
+
+test('will not invoke until it has stopped being called for N milliseconds', function (t) {
```
Comment (angus-c) [inline test/function-debounce/index.js]: Can you add an additional test with 0 as the timeout arg?

Comment (ETNOL) [inline test/function-debounce/index.js]: Added test and it failed =P. Added an escape conditional to invoke immediately if wait time passed is equal to 0.

**Comment by angus-c** [comment]
landing but will make some changes

**Pull request merged**